internal/kafka/internal/workers/kafka_mgrs/kafkas_mgr.go

```diff
@@ -120,6 +120,37 @@ func (k *KafkaManager) Reconcile() []error {
 		return encounteredErrors
 	}
 
+	for _, kafka := range kafkas {
+		if !kafka.CanBeSuspended() {
+			// this kafka is not in a state that can be suspended
+			continue
+		}
+
+		expired, residualLifespan := kafka.IsExpired()
+		if expired {
+			// Expired kafkas will be deleted elsewhere
+			continue
+		}
+
+		// get the billing model
+		bm, err := k.kafkaConfig.GetBillingModelByID(kafka.InstanceType, kafka.ActualKafkaBillingModel)
+		if err != nil {
+			wrappedError := errors.Wrap(err, "failed to suspend expired Kafka instances")
+			encounteredErrors = append(encounteredErrors, wrappedError)
+			continue
+		}
+
+		if !residualLifespan.Infinite() && residualLifespan.Days() < float64(bm.GracePeriodDays) {
+			// the instance is in grace period
+			_, err := k.kafkaService.UpdateStatus(kafka.ID, constants.KafkaRequestStatusSuspending)
```

Comment:
Isn't there a possibility that a developer kafka will be seen in `suspsending` state just before (a few nanoseconds) before it moves to deprovision?

Reply:
Can you elaborate a bit more? I'm not sure I get what you mean.

Reply:
The loop seems to be treating all the Kafkas the same irrespective of their instance type. 
What I am asking is, would it be possible (and acceptable?) that a developer cluster that's going to expire (in a few nanoseconds i.e just after line 129 is evaluated) to be have the status of `suspending`?

Reply:
The developer instance has a grace period set to 0 (zero). It won't enter the suspending state: it will be deleted immediately upon expiration.

Reply:
Thank you.